### PR TITLE
feat: add Wazuh 4.8 support, extend functionality, add tests

### LINKED ISSUE #6: Support for Wazuh >=4.8 (Opensearch API for vulnerabilities)
## Description
Currently, the exporter relies on the old Wazuh API endpoint `/vulnerability` to fetch vulnerability findings.
Starting from **Wazuh 4.8**, this endpoint has been removed.
Vulnerability data is now only available via **Wazuh Indexer (Opensearch/Elasticsearch)** in the index:

```
wazuh-states-vulnerabilities-*
```

## Problem

* The script no longer works on Wazuh 4.8+ since `/vulnerability` API is gone.

## Request

* Add support for Wazuh 4.8+ by pulling vulnerability data from **Opensearch API**.

## Query example

```
GET /wazuh-states-vulnerabilities-*/_search
{
  "query": {
    "bool": {
      "should": [
        {"match": {"vulnerability.severity": "Critical"}},
        {"match": {"vulnerability.severity": "High"}},
        {"match": {"vulnerability.severity": "Medium"}},
        {"match": {"vulnerability.severity": "Low"}},
        {"match": {"vulnerability.severity": ""}}
      ],
      "minimum_should_match": 1
    }
  }
}
```

## JSON example

```json
{
    "took": 8,
    "timed_out": false,
    "_shards": {
      "total": 1,
      "successful": 1,
      "skipped": 0,
      "failed": 0
    },
    "hits": {
      "total": {
        "value": 125,
        "relation": "eq"
      },
      "max_score": 5.596354,
      "hits": [
        {
          "_index": "wazuh-states-vulnerabilities-wazuh-server",
          "_id": "001_b5f8c1a3d7e902b4c6d8e0f2a4b6c8d0e2f4a6b8_CVE-2025-27558",
          "_score": 5.596354,
          "_source": {
            "agent": {
              "id": "001",
              "name": "myhost0",
              "type": "Wazuh",
              "version": "v4.11.1"
            },
            "host": {
              "os": {
                "full": "Ubuntu 24.04.2 LTS (Noble Numbat)",
                "kernel": "6.8.0-62-generic",
                "name": "Ubuntu",
                "platform": "ubuntu",
                "type": "ubuntu",
                "version": "24.04.2"
              }
            },
            "package": {
              "architecture": "amd64",
              "description": "Signed kernel image generic",
              "name": "linux-image-6.8.0-60-generic",
              "size": 15025152,
              "type": "deb",
              "version": "6.8.0-60.63"
            },
            "vulnerability": {
              "category": "Packages",
              "classification": "-",
              "description": "IEEE P802.11-REVme D1.1 through D7.0 allows FragAttacks against meshnetworks. In mesh networks using Wi-Fi Protected Access (WPA, WPA2, orWPA3) or Wired Equivalent Privacy (WEP), an adversary can exploit thisvulnerability to inject arbitrary frames towards devices that supportreceiving non-SSP A-MSDU frames. NOTE: this issue exists because of anincorrect fix for CVE-2020-24588. P802.11-REVme, as of early 2025, is aplanned release of the 802.11 standard.",
              "detected_at": "2025-06-30T17:07:15.204Z",
              "enumeration": "CVE",
              "id": "CVE-2025-27558",
              "published_at": "2025-05-21T19:16:08Z",
              "reference": "https://ubuntu.com/security/CVE-2025-27558, https://www.cve.org/CVERecord?id=CVE-2025-27558",
              "scanner": {
                "condition": "Package default status",
                "reference": "https://cti.wazuh.com/vulnerabilities/cves/CVE-2025-27558",
                "source": "Canonical Security Tracker",
                "vendor": "Wazuh"
              },
              "score": {
                "base": 9.1,
                "version": "3.1"
              },
              "severity": "Critical",
              "under_evaluation": false
            },
            "wazuh": {
              "cluster": {
                "name": "wazuh-server"
              },
              "schema": {
                "version": "1.0.0"
              }
            }
          }
        },
        {
          "_index": "wazuh-states-vulnerabilities-wazuh-server",
          "_id": "001_b5f8c1a3d7e902b4c6d8e0f2a4b6c8d0e2f4a6b8_CVE-2024-56180",
          "_score": 5.596354,
          "_source": {
            "agent": {
              "id": "001",
              "name": "myhost0",
              "type": "Wazuh",
              "version": "v4.11.1"
            },
            "host": {
              "os": {
                "full": "Ubuntu 24.04.2 LTS (Noble Numbat)",
                "kernel": "6.8.0-62-generic",
                "name": "Ubuntu",
                "platform": "ubuntu",
                "type": "ubuntu",
                "version": "24.04.2"
              }
            },
            "package": {
              "architecture": "amd64",
              "description": "Signed kernel image generic",
              "name": "linux-image-6.8.0-60-generic",
              "size": 15025152,
              "type": "deb",
              "version": "6.8.0-60.63"
            },
            "vulnerability": {
              "category": "Packages",
              "classification": "-",
              "description": "CWE-502 Deserialization of Untrusted Data at theeventmesh-meta-raft plugin module in Apache EventMesh master branch withoutrelease version on windows\\linux\\mac os e.g. platforms allows attackers tosend controlled message and remote code execute via hessian deserializationrpc protocol. Users can use the code under the master branch in projectrepo or version 1.11.0 to fix this issue.",
              "detected_at": "2025-06-30T17:07:15.217Z",
              "enumeration": "CVE",
              "id": "CVE-2024-56180",
              "published_at": "2025-02-14T14:15:32Z",
              "reference": "https://ubuntu.com/security/CVE-2024-56180, https://www.cve.org/CVERecord?id=CVE-2024-56180",
              "scanner": {
                "condition": "Package default status",
                "reference": "https://cti.wazuh.com/vulnerabilities/cves/CVE-2024-56180",
                "source": "Canonical Security Tracker",
                "vendor": "Wazuh"
              },
              "score": {
                "base": 9.8,
                "version": "3.1"
              },
              "severity": "Critical",
              "under_evaluation": false
            },
            "wazuh": {
              "cluster": {
                "name": "wazuh-server"
              },
              "schema": {
                "version": "1.0.0"
              }
            }
          }
        },
        {
          "_index": "wazuh-states-vulnerabilities-wazuh-server",
          "_id": "001_b5f8c1a3d7e902b4c6d8e0f2a4b6c8d0e2f4a6b8_CVE-2021-3773",
          "_score": 5.596354,
          "_source": {
            "agent": {
              "id": "001",
              "name": "myhost0",
              "type": "Wazuh",
              "version": "v4.11.1"
            },
            "host": {
              "os": {
                "full": "Ubuntu 24.04.2 LTS (Noble Numbat)",
                "kernel": "6.8.0-62-generic",
                "name": "Ubuntu",
                "platform": "ubuntu",
                "type": "ubuntu",
                "version": "24.04.2"
              }
            },
            "package": {
              "architecture": "amd64",
              "description": "Signed kernel image generic",
              "name": "linux-image-6.8.0-60-generic",
              "size": 15025152,
              "type": "deb",
              "version": "6.8.0-60.63"
            },
            "vulnerability": {
              "category": "Packages",
              "classification": "-",
              "description": "A flaw in netfilter could allow a network-connected attacker to inferopenvpn connection endpoint information for further use in traditionalnetwork attacks.",
              "detected_at": "2025-06-30T17:07:15.218Z",
              "enumeration": "CVE",
              "id": "CVE-2021-3773",
              "published_at": "2022-02-16T19:15:08Z",
              "reference": "https://ubuntu.com/security/CVE-2021-3773, https://www.cve.org/CVERecord?id=CVE-2021-3773",
              "scanner": {
                "condition": "Package default status",
                "reference": "https://cti.wazuh.com/vulnerabilities/cves/CVE-2021-3773",
                "source": "Canonical Security Tracker",
                "vendor": "Wazuh"
              },
              "score": {
                "base": 9.8,
                "version": "3.1"
              },
              "severity": "Critical",
              "under_evaluation": false
            },
            "wazuh": {
              "cluster": {
                "name": "wazuh-server"
              },
              "schema": {
                "version": "1.0.0"
              }
            }
          }
        },
        {
          "_index": "wazuh-states-vulnerabilities-wazuh-server",
          "_id": "001_b5f8c1a3d7e902b4c6d8e0f2a4b6c8d0e2f4a6b8_CVE-2025-27558",
          "_score": 5.596354,
          "_source": {
            "agent": {
              "id": "001",
              "name": "myhost0",
              "type": "Wazuh",
              "version": "v4.11.1"
            },
            "host": {
              "os": {
                "full": "Ubuntu 24.04.2 LTS (Noble Numbat)",
                "kernel": "6.8.0-62-generic",
                "name": "Ubuntu",
                "platform": "ubuntu",
                "type": "ubuntu",
                "version": "24.04.2"
              }
            },
            "package": {
              "architecture": "amd64",
              "description": "Signed kernel image generic",
              "name": "linux-image-6.8.0-62-generic",
              "size": 15033344,
              "type": "deb",
              "version": "6.8.0-62.65"
            },
            "vulnerability": {
              "category": "Packages",
              "classification": "-",
              "description": "IEEE P802.11-REVme D1.1 through D7.0 allows FragAttacks against meshnetworks. In mesh networks using Wi-Fi Protected Access (WPA, WPA2, orWPA3) or Wired Equivalent Privacy (WEP), an adversary can exploit thisvulnerability to inject arbitrary frames towards devices that supportreceiving non-SSP A-MSDU frames. NOTE: this issue exists because of anincorrect fix for CVE-2020-24588. P802.11-REVme, as of early 2025, is aplanned release of the 802.11 standard.",
              "detected_at": "2025-06-30T17:07:15.926Z",
              "enumeration": "CVE",
              "id": "CVE-2025-27558",
              "published_at": "2025-05-21T19:16:08Z",
              "reference": "https://ubuntu.com/security/CVE-2025-27558, https://www.cve.org/CVERecord?id=CVE-2025-27558",
              "scanner": {
                "condition": "Package default status",
                "reference": "https://cti.wazuh.com/vulnerabilities/cves/CVE-2025-27558",
                "source": "Canonical Security Tracker",
                "vendor": "Wazuh"
              },
              "score": {
                "base": 9.1,
                "version": "3.1"
              },
              "severity": "Critical",
              "under_evaluation": false
            },
            "wazuh": {
              "cluster": {
                "name": "wazuh-server"
              },
              "schema": {
                "version": "1.0.0"
              }
            }
          }
        },
        {
          "_index": "wazuh-states-vulnerabilities-wazuh-server",
          "_id": "001_b5f8c1a3d7e902b4c6d8e0f2a4b6c8d0e2f4a6b8_CVE-2024-56180",
          "_score": 5.596354,
          "_source": {
            "agent": {
              "id": "001",
              "name": "myhost0",
              "type": "Wazuh",
              "version": "v4.11.1"
            },
            "host": {
              "os": {
                "full": "Ubuntu 24.04.2 LTS (Noble Numbat)",
                "kernel": "6.8.0-62-generic",
                "name": "Ubuntu",
                "platform": "ubuntu",
                "type": "ubuntu",
                "version": "24.04.2"
              }
            },
            "package": {
              "architecture": "amd64",
              "description": "Signed kernel image generic",
              "name": "linux-image-6.8.0-62-generic",
              "size": 15033344,
              "type": "deb",
              "version": "6.8.0-62.65"
            },
            "vulnerability": {
              "category": "Packages",
              "classification": "-",
              "description": "CWE-502 Deserialization of Untrusted Data at theeventmesh-meta-raft plugin module in Apache EventMesh master branch withoutrelease version on windows\\linux\\mac os e.g. platforms allows attackers tosend controlled message and remote code execute via hessian deserializationrpc protocol. Users can use the code under the master branch in projectrepo or version 1.11.0 to fix this issue.",
              "detected_at": "2025-06-30T17:07:15.934Z",
              "enumeration": "CVE",
              "id": "CVE-2024-56180",
              "published_at": "2025-02-14T14:15:32Z",
              "reference": "https://ubuntu.com/security/CVE-2024-56180, https://www.cve.org/CVERecord?id=CVE-2024-56180",
              "scanner": {
                "condition": "Package default status",
                "reference": "https://cti.wazuh.com/vulnerabilities/cves/CVE-2024-56180",
                "source": "Canonical Security Tracker",
                "vendor": "Wazuh"
              },
              "score": {
                "base": 9.8,
                "version": "3.1"
              },
              "severity": "Critical",
              "under_evaluation": false
            },
            "wazuh": {
              "cluster": {
                "name": "wazuh-server"
              },
              "schema": {
                "version": "1.0.0"
              }
            }
          }
        },
        {
          "_index": "wazuh-states-vulnerabilities-wazuh-server",
          "_id": "001_b5f8c1a3d7e902b4c6d8e0f2a4b6c8d0e2f4a6b8_CVE-2021-3773",
          "_score": 5.596354,
          "_source": {
            "agent": {
              "id": "001",
              "name": "myhost0",
              "type": "Wazuh",
              "version": "v4.11.1"
            },
            "host": {
              "os": {
                "full": "Ubuntu 24.04.2 LTS (Noble Numbat)",
                "kernel": "6.8.0-62-generic",
                "name": "Ubuntu",
                "platform": "ubuntu",
                "type": "ubuntu",
                "version": "24.04.2"
              }
            },
            "package": {
              "architecture": "amd64",
              "description": "Signed kernel image generic",
              "name": "linux-image-6.8.0-62-generic",
              "size": 15033344,
              "type": "deb",
              "version": "6.8.0-62.65"
            },
            "vulnerability": {
              "category": "Packages",
              "classification": "-",
              "description": "A flaw in netfilter could allow a network-connected attacker to inferopenvpn connection endpoint information for further use in traditionalnetwork attacks.",
              "detected_at": "2025-06-30T17:07:15.935Z",
              "enumeration": "CVE",
              "id": "CVE-2021-3773",
              "published_at": "2022-02-16T19:15:08Z",
              "reference": "https://ubuntu.com/security/CVE-2021-3773, https://www.cve.org/CVERecord?id=CVE-2021-3773",
              "scanner": {
                "condition": "Package default status",
                "reference": "https://cti.wazuh.com/vulnerabilities/cves/CVE-2021-3773",
                "source": "Canonical Security Tracker",
                "vendor": "Wazuh"
              },
              "score": {
                "base": 9.8,
                "version": "3.1"
              },
              "severity": "Critical",
              "under_evaluation": false
            },
            "wazuh": {
              "cluster": {
                "name": "wazuh-server"
              },
              "schema": {
                "version": "1.0.0"
              }
            }
          }
        },
        {
          "_index": "wazuh-states-vulnerabilities-wazuh-server",
          "_id": "002_b5f8c1a3d7e902b4c6d8e0f2a4b6c8d0e2f4a6b8_CVE-2022-44640",
          "_score": 5.596354,
          "_source": {
            "agent": {
              "id": "002",
              "name": "myhost1",
              "type": "Wazuh",
              "version": "v4.11.1"
            },
            "host": {
              "os": {
                "full": "Ubuntu 24.04.1 LTS (Noble Numbat)",
                "kernel": "6.8.0-62-generic",
                "name": "Ubuntu",
                "platform": "ubuntu",
                "type": "ubuntu",
                "version": "24.04.1"
              }
            },
            "package": {
              "architecture": "amd64",
              "description": "Heimdal Kerberos - Base library",
              "name": "libheimbase1t64-heimdal",
              "size": 95232,
              "type": "deb",
              "version": "7.8.git20221117.28daf24+dfsg-5ubuntu3"
            },
            "vulnerability": {
              "category": "Packages",
              "classification": "-",
              "description": "Heimdal before 7.7.1 allows remote attackers to execute arbitrary codebecause of an invalid free in the ASN.1 codec used by the Key DistributionCenter (KDC).",
              "detected_at": "2025-06-30T17:07:16.011Z",
              "enumeration": "CVE",
              "id": "CVE-2022-44640",
              "published_at": "2022-12-25T05:15:11Z",
              "reference": "https://ubuntu.com/security/CVE-2022-44640, https://ubuntu.com/security/notices/USN-5800-1, https://www.cve.org/CVERecord?id=CVE-2022-44640",
              "scanner": {
                "condition": "Package default status",
                "reference": "https://cti.wazuh.com/vulnerabilities/cves/CVE-2022-44640",
                "source": "Canonical Security Tracker",
                "vendor": "Wazuh"
              },
              "score": {
                "base": 9.8,
                "version": "3.1"
              },
              "severity": "Critical",
              "under_evaluation": false
            },
            "wazuh": {
              "cluster": {
                "name": "wazuh-server"
              },
              "schema": {
                "version": "1.0.0"
              }
            }
          }
        },
        {
          "_index": "wazuh-states-vulnerabilities-wazuh-server",
          "_id": "002_b5f8c1a3d7e902b4c6d8e0f2a4b6c8d0e2f4a6b8_CVE-2025-27558",
          "_score": 5.596354,
          "_source": {
            "agent": {
              "id": "002",
              "name": "myhost1",
              "type": "Wazuh",
              "version": "v4.11.1"
            },
            "host": {
              "os": {
                "full": "Ubuntu 24.04.1 LTS (Noble Numbat)",
                "kernel": "6.8.0-62-generic",
                "name": "Ubuntu",
                "platform": "ubuntu",
                "type": "ubuntu",
                "version": "24.04.1"
              }
            },
            "package": {
              "architecture": "amd64",
              "description": "Signed kernel image generic",
              "name": "linux-image-6.8.0-60-generic",
              "size": 15025152,
              "type": "deb",
              "version": "6.8.0-60.63"
            },
            "vulnerability": {
              "category": "Packages",
              "classification": "-",
              "description": "IEEE P802.11-REVme D1.1 through D7.0 allows FragAttacks against meshnetworks. In mesh networks using Wi-Fi Protected Access (WPA, WPA2, orWPA3) or Wired Equivalent Privacy (WEP), an adversary can exploit thisvulnerability to inject arbitrary frames towards devices that supportreceiving non-SSP A-MSDU frames. NOTE: this issue exists because of anincorrect fix for CVE-2020-24588. P802.11-REVme, as of early 2025, is aplanned release of the 802.11 standard.",
              "detected_at": "2025-06-30T17:07:16.927Z",
              "enumeration": "CVE",
              "id": "CVE-2025-27558",
              "published_at": "2025-05-21T19:16:08Z",
              "reference": "https://ubuntu.com/security/CVE-2025-27558, https://www.cve.org/CVERecord?id=CVE-2025-27558",
              "scanner": {
                "condition": "Package default status",
                "reference": "https://cti.wazuh.com/vulnerabilities/cves/CVE-2025-27558",
                "source": "Canonical Security Tracker",
                "vendor": "Wazuh"
              },
              "score": {
                "base": 9.1,
                "version": "3.1"
              },
              "severity": "Critical",
              "under_evaluation": false
            },
            "wazuh": {
              "cluster": {
                "name": "wazuh-server"
              },
              "schema": {
                "version": "1.0.0"
              }
            }
          }
        },
        {
          "_index": "wazuh-states-vulnerabilities-wazuh-server",
          "_id": "002_b5f8c1a3d7e902b4c6d8e0f2a4b6c8d0e2f4a6b8_CVE-2024-56180",
          "_score": 5.596354,
          "_source": {
            "agent": {
              "id": "002",
              "name": "myhost1",
              "type": "Wazuh",
              "version": "v4.11.1"
            },
            "host": {
              "os": {
                "full": "Ubuntu 24.04.1 LTS (Noble Numbat)",
                "kernel": "6.8.0-62-generic",
                "name": "Ubuntu",
                "platform": "ubuntu",
                "type": "ubuntu",
                "version": "24.04.1"
              }
            },
            "package": {
              "architecture": "amd64",
              "description": "Signed kernel image generic",
              "name": "linux-image-6.8.0-60-generic",
              "size": 15025152,
              "type": "deb",
              "version": "6.8.0-60.63"
            },
            "vulnerability": {
              "category": "Packages",
              "classification": "-",
              "description": "CWE-502 Deserialization of Untrusted Data at theeventmesh-meta-raft plugin module in Apache EventMesh master branch withoutrelease version on windows\\linux\\mac os e.g. platforms allows attackers tosend controlled message and remote code execute via hessian deserializationrpc protocol. Users can use the code under the master branch in projectrepo or version 1.11.0 to fix this issue.",
              "detected_at": "2025-06-30T17:07:16.937Z",
              "enumeration": "CVE",
              "id": "CVE-2024-56180",
              "published_at": "2025-02-14T14:15:32Z",
              "reference": "https://ubuntu.com/security/CVE-2024-56180, https://www.cve.org/CVERecord?id=CVE-2024-56180",
              "scanner": {
                "condition": "Package default status",
                "reference": "https://cti.wazuh.com/vulnerabilities/cves/CVE-2024-56180",
                "source": "Canonical Security Tracker",
                "vendor": "Wazuh"
              },
              "score": {
                "base": 9.8,
                "version": "3.1"
              },
              "severity": "Critical",
              "under_evaluation": false
            },
            "wazuh": {
              "cluster": {
                "name": "wazuh-server"
              },
              "schema": {
                "version": "1.0.0"
              }
            }
          }
        },
        {
          "_index": "wazuh-states-vulnerabilities-wazuh-server",
          "_id": "002_b5f8c1a3d7e902b4c6d8e0f2a4b6c8d0e2f4a6b8_CVE-2021-3773",
          "_score": 5.596354,
          "_source": {
            "agent": {
              "id": "002",
              "name": "myhost1",
              "type": "Wazuh",
              "version": "v4.11.1"
            },
            "host": {
              "os": {
                "full": "Ubuntu 24.04.1 LTS (Noble Numbat)",
                "kernel": "6.8.0-62-generic",
                "name": "Ubuntu",
                "platform": "ubuntu",
                "type": "ubuntu",
                "version": "24.04.1"
              }
            },
            "package": {
              "architecture": "amd64",
              "description": "Signed kernel image generic",
              "name": "linux-image-6.8.0-60-generic",
              "size": 15025152,
              "type": "deb",
              "version": "6.8.0-60.63"
            },
            "vulnerability": {
              "category": "Packages",
              "classification": "-",
              "description": "A flaw in netfilter could allow a network-connected attacker to inferopenvpn connection endpoint information for further use in traditionalnetwork attacks.",
              "detected_at": "2025-06-30T17:07:16.938Z",
              "enumeration": "CVE",
              "id": "CVE-2021-3773",
              "published_at": "2022-02-16T19:15:08Z",
              "reference": "https://ubuntu.com/security/CVE-2021-3773, https://www.cve.org/CVERecord?id=CVE-2021-3773",
              "scanner": {
                "condition": "Package default status",
                "reference": "https://cti.wazuh.com/vulnerabilities/cves/CVE-2021-3773",
                "source": "Canonical Security Tracker",
                "vendor": "Wazuh"
              },
              "score": {
                "base": 9.8,
                "version": "3.1"
              },
              "severity": "Critical",
              "under_evaluation": false
            },
            "wazuh": {
              "cluster": {
                "name": "wazuh-server"
              },
              "schema": {
                "version": "1.0.0"
              }
            }
          }
        }
      ]
    }
  }
```

### PR DESCRIPTION
**Pull request by Haralishev77**
## Update Wazuh Exporter with OpenSearch support (Wazuh version >=4.8), streaming split output, and expanded tests

This PR adds full support for Wazuh API 4.8+ via OpenSearch, introduces streaming split output, improves documentation and the example usage, and significantly expands test coverage. Backward compatibility with pre‑4.8 Wazuh is preserved.
It addresses the following issue: [#6](https://github.com/quirinziessler/wazuh_findings_exporter/issues/6)

## Changes
- Added OpenSearch integration for Wazuh 4.8+ and streaming split output with scroll handling
- Added `output_mode` validation and improved error handling for OpenSearch operations
- Expanded unit tests to cover all branches and edge cases in the importer logic
- Added `tox` configuration to run tests across 3.10-3.14 Python versions
- Removed support for Python versions earlier than 3.10
- Enhanced logging and error handling throughout the Wazuh importer
- Updated type hints and documentation for public methods

### CODE REVIEW

**Comment by Haralishev77** [comment]
Hi @quirinziessler,

I'd like to propose moving the Wazuh Exporter to version 2.0 in order to simplify and refactor the code. The update would maintain backward compatibility with Wazuh API < 4.8.

I’d appreciate your feedback on a few points:

- Do you agree with the version bump to 2.0 to reflect code simplification and refactoring?
- Are there any parts of the current implementation you would like to preserve or adjust?
- Anything you’d like to add, remove, or keep as-is before finalizing the 2.0 release?

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,5 @@
 urllib3==2.5.0
-requests==2.32.4
\ No newline at end of file
+requests==2.32.4
+coloredlogs==15.0.1
+opensearch-py==3.0.0
+opensearch_dsl==2.1.0
```
Comment (quirinziessler) [inline requirements.txt]: I can't see `opensearch_dsl` anywhere being loaded within the project. Is it realy required? The coloredlogs thingy is also smth I wouldn't get for myself but okay I will accept it.

Comment (quirinziessler) [reply requirements.txt]: btw also retired - so not usable

**Comment by Haralishev77** [comment]
Thanks a lot for the review, @quirinziessler! I’ll continue working on this once I find some time

**Comment by Haralishev77** [comment]
Hi @quirinziessler,
I’ve finally finished this.

For Wazuh versions >= 4.8, I added the ability to split the output. I tested the script with 2 million hints, and the resulting JSON grew to about 6 GB, which is impractical.

**Comment by Haralishev77** [comment]
Also, we should update [defectdojo page](https://docs.defectdojo.com/supported_tools/parsers/file/wazuh/) with a new example of the acceptable JSON format for Wazuh >= 4.8.

**Comment by Haralishev77** [comment]
Hi @quirinziessler,
I’ve fixed the issues mentioned above. Could you please request another review?

**Comment by Haralishev77** [comment]
Hi, @quirinziessler! Could you review this, please?

**Comment by Haralishev77** [comment]
I can also add CI tests and linters if needed. In another PR

**Comment by Haralishev77** [comment]
@quirinziessler up

**Comment by quirinziessler** [comment]
thx @Haralishev77! Merged it. 

Btw if you want, you can also take over this project. I am atm not using Wazuh productive and so have less use on this. Any interest?

**Comment by Haralishev77** [comment]
Hello, @quirinziessler!

Sure, I’d be happy to take it over and continue developing it for the community

I have several ideas I'd like to implement

**Comment by Haralishev77** [comment]
Hi, @quirinziessler! 
What do we do next with the project?

**Comment by quirinziessler** [comment]
Hi @Haralishev77 - sorry I didn't get back to you earlier. As I said I am happy to pass this project further to you, if you want.

Could you please once update your fork to the latest? Then I will add a note to my repo and archive it.

**Comment by Haralishev77** [comment]
I've updated my fork and will convert it into a standalone repository.

Could you please let me know how we should proceed with the PyPI package?
If possible, I would prefer to be added as a maintainer so future releases can remain under the same package name

@quirinziessler

**Comment by Haralishev77** [comment]
Also, there is one more option: we could transfer the repository ownership. That way the history, stars, and issues would be preserved, and the project would continue seamlessly under my account. But idk, is that ok for you?

There is no need to archive in that case and the old link will redirect automatically

**Comment by quirinziessler** [comment]
Second is totally fine for me and as you said probably the easiest. Send you an invite for pypi. I will also send a request for transfer of the ownership once you renamed your fork.

And sorry I didn't read further in the first place and already pushed a note to the ReadMe. Pls remove it once transfer is done.

**Comment by Haralishev77** [comment]
@quirinziessler 
I've accepted the invitation for pypi and renamed my fork. 
Ok, I will check README

**Comment by quirinziessler** [comment]
Seems like it worked. 😀 Thx for continuing the project and if you need input feel free to ping me!

**Comment by Haralishev77** [comment]
Sure, thank you!
